scheduler/core/components/optimizer/dummy.py

```diff
@@ -58,7 +58,7 @@ def add(self, group: GroupData, plans: Plans, interval: Optional[Interval] = Non
         for observation in group.group.observations():
             plan = plans[observation.site]
             if not plan.is_full and plan.site == observation.site:
-                obs_len = plan.time2slots(observation.exec_time())
+                obs_len = plan.time2slots(plan.time_slot_length, observation.exec_time())
```

Comment:
Why not using self then?

Reply:
In GreedyMax, Bryan is doing this:
```
n_slots_remaining = Plan.time2slots(self.time_slot_length, time_remaining)
```
and doesn't have a `Plan` instance at that point. He was passing in `GreedyMaxOptimizer` since it has a `time_slot_length` member, so via duck typing, everything was working, but it just seemed wrong to me to fudge a `self` (which should always be a `Plan`) with a `GreedyMaxOptimizer` (which happened to have a `time_slot_length` member, so it worked).

I figured it would cause us a huge headache with mypy and typing in general.

Reply:
Not sure how we should handle this... any ideas? Maybe it should be an independent function that just takes a `time_slot_length` and a `time_delta`. That way, we can eliminate it from `Plan` and still use it where needed, and use it from `GreedyMax` in the way that Bryan is using it.